### PR TITLE
repo: change logic for creation vs package creation

### PR DESCRIPTION
**Pull request by larshp**
creating new repo will now check the package exists

ie. responsibility for creating the package is/can be moved to the user interface/frontend, instead of backend logic

@mbtools I think this reverts the behavior of some PR?

error is now given when a non-existing package is chosen,
image

### CODE REVIEW

**Comment by mbtools** [comment]
Why? What's wrong with the existing behaviour?

**Comment by larshp** [comment]
a: now the popup can be moved to the UI, instead of nested in the application logic
b: consistency, if creating a repo where the remote does not have any packages, then it shows a popup, else not
c: its not possible to determine what the user wants without asking the user, eg. superclass + transport layer + software component + description

**Comment by larshp** [comment]
found the PR, https://github.com/abapGit/abapGit/pull/4431

it currently works this way in order to handle namespaced packages?

File: src/ui/routing/zcl_abapgit_services_repo.clas.abap [hunk 1]
```diff
@@ -180,22 +174,6 @@ CLASS zcl_abapgit_services_repo IMPLEMENTATION.
...
-  METHOD check_and_create_package.
-
-    IF zcl_abapgit_factory=>get_sap_package( iv_package )->exists( ) = abap_false.
-      " Check if any package is included in remote
-      READ TABLE it_remote TRANSPORTING NO FIELDS
-        WITH KEY file
-        COMPONENTS filename = zcl_abapgit_filename_logic=>c_package_file.
-      IF sy-subrc <> 0.
-        " If not, prompt to create it
-        create_package( iv_package ).
```
Comment (larshp) [inline src/ui/routing/zcl_abapgit_services_repo.clas.abap:192]: how about, here instead of showing the popup, throw an exception? @mbtools

Comment (mbtools) [reply src/ui/routing/zcl_abapgit_services_repo.clas.abap:192]: Don't have time to look at it now. There should not be a popup (b-case) but an error to fallback to the "new repo" form.

**Comment by mbtools** [comment]
In many cases, the defaults are completely sufficient. Enter a $package, ok, pull. Or "ZTEST", ok, pull. That was the main point of it.

If the defaults are not ok, the user always has the choice to create the package first. 

https://docs.abapgit.org/user-guide/reference/packages.html

**Comment by larshp** [comment]
moving to https://github.com/abapGit/abapGit/pull/6626

**Pull request closed without merging**